starters/next-drupal-starter/pages/examples/ssg-isr.jsx

```diff
@@ -0,0 +1,101 @@
+import { NextSeo } from "next-seo";
+import { isMultiLanguage } from "../../lib/isMultiLanguage";
+import {
+  getCurrentLocaleStore,
+  globalDrupalStateStores,
+} from "../../lib/drupalStateContext";
+
+import { ArticleGridItem, withGrid } from "../../components/grid";
+import Layout from "../../components/layout";
+
+export default function HomepageTemplate({
+  articles,
+  footerMenu,
+  hrefLang,
+  multiLanguage,
+}) {
+  const HomepageHeader = () => (
+    <div className="prose sm:prose-xl mt-20 flex flex-col mx-auto max-w-fit">
+      <h1 className="prose text-4xl text-center h-full">
+        Articles
+      </h1>
+    </div>
+  );
+
+  const ArticleGrid = withGrid(ArticleGridItem);
+
+  return (
+    <Layout footerMenu={footerMenu}>
+      <NextSeo
+        title="Decoupled Next Drupal Demo"
+        description="Generated by create next app."
+        languageAlternates={hrefLang || false}
+      />
+      <>
+        <HomepageHeader />
+        <div className="mt-8 prose lg:prose-xl max-w-lg mx-auto lg:grid-cols-3 lg:max-w-screen-lg">
+          <p>
+            <em>By default, this starter kit is optimized for SSR and Edge
+              Caching on Pantheon. This example instead uses Incremental Static
+              Regeneration and is provided as a reference for cases where Next.js
+              static generation options would be beneficial.</em>
+          </p>
+        </div>
+        <section>
+          <ArticleGrid
+            data={articles}
+            contentType="articles"
+            multiLanguage={multiLanguage}
+          />
+        </section>
+      </>
+    </Layout>
+  );
+}
+
+export async function getStaticProps(context) {
+  const origin = process.env.NEXT_PUBLIC_FRONTEND_URL;
+  const { locales } = context;
+  // if there is more than one language in context.locales,
+  // assume multilanguage is enabled.
+  const multiLanguage = isMultiLanguage(locales);
+  const hrefLang = locales.map((locale) => {
+    return {
+      hrefLang: locale,
+      href: origin + "/" + locale,
+    };
+  });
+
+  try {
+    const store = getCurrentLocaleStore(
+      context.locale,
+      globalDrupalStateStores
+    );
+
+    const articles = await store.getObject({
+      objectName: "node--article",
+      params: "include=field_media_image.field_media_image",
+    });
+
+    if (!articles) {
+      throw new Error(
+        "No articles returned. Make sure the objectName and params are valid!"
+      );
+    }
+
+    const footerMenu = await store.getObject({
+      objectName: "menu_items--main",
+    });
+
+    return {
+      props: { articles, hrefLang, multiLanguage, footerMenu },
+      revalidate: 60,
```

Comment:
When I do a build, this page shows up with the SSG indicator but not the `(ISR: 60 seconds)` part that I would expect here.

```
├ ● /examples/ssg-isr                                              2.44 kB         176 kB
```

And it seems to have the behavior we observed in the past where a content change IS being recognized after the first 60 seconds, but subsequent changes seem to NOT run `getStaticProps` again.

Reply:
I see what you mean about the output, but with the most recent round of changes I've confirmed that ISR is in fact working. Needed to use refresh: true for all calls here as well. I know that isn't ideal for this SSR use case, but I actually think it is a powerful feature that Drupal State can persist for the life of the node process here (and I'm still surprised honestly.) If we can give users better control over that (and maybe re-evaluate what the default should be?) I think this is a good thing long term.

Reply:
I am seeing the content updates now past the first edit, however what is curious to me is that the new content is available before the `revalidate` time has elapsed. I would not expect the new content to be available until 60 seconds pass, but I can see it right away when I refresh the page?

If I change the content and navigate around the SSR'd pages then navigate back to the ISR page, I DO NOT see the new content until I manually refresh the page. Just a weird quirk of how ISR works I guess?

The only issue is it can lead to a state where I see the fresh content on the SSR pages and stale content on the ISR page (until I refresh)

Not much to do about this I think, just the way it works. (Also wanted to drop these thoughts here because someone may log this as a bug...)